File: FlowRegistry/src/main/java/com/boraydata/flowregistry/utils/UserUtils.java
package com.boraydata.flowregistry.utils;

import com.boraydata.flowregistry.entity.User;
import org.springframework.security.core.context.SecurityContext;
import org.springframework.security.web.context.HttpSessionSecurityContextRepository;
import org.springframework.web.context.request.RequestContextHolder;
import org.springframework.web.context.request.ServletRequestAttributes;

public class UserUtils {

    /**
     * Get current User from Security
     */
    public static User currentUser() {
        ServletRequestAttributes attributes = (ServletRequestAttributes) RequestContextHolder.getRequestAttributes();
        SecurityContext attribute =
                (SecurityContext) attributes.getRequest().getSession().getAttribute(HttpSessionSecurityContextRepository.SPRING_SECURITY_CONTEXT_KEY);
        if (attribute == null) {
            return null;
        }
        if (attribute.getAuthentication() == null) {
            return null;
        }
        return (User) attribute.getAuthentication().getPrincipal();
    }
}

File: FlowStudio/src/main/java/com/boraydata/workflow/utils/UserUtils.kt
package com.boraydata.workflow.utils

import com.boraydata.workflow.entity.User
import org.springframework.security.core.context.SecurityContext
import org.springframework.security.web.context.HttpSessionSecurityContextRepository
import org.springframework.web.context.request.RequestContextHolder
import org.springframework.web.context.request.ServletRequestAttributes

object UserUtils {

    /**
     * Get current User from Security
     * */
    fun currentUser(): User {
        val attributes = RequestContextHolder.getRequestAttributes() as ServletRequestAttributes
        val attribute =
            attributes.request.session.getAttribute(HttpSessionSecurityContextRepository.SPRING_SECURITY_CONTEXT_KEY) as SecurityContext
        return attribute.authentication.principal as User
    }
}